### PR TITLE
Removing dead code in gpMgmt bash libraries

### PR DESCRIPTION
**Pull request by danielgustafsson**
While in there looking at other things, realized that there was quite a lot of dead/unused code in the gpMgmt bin/lib bash files. This mostly removes dead code, but also a bit of unused functionality like printing the version (which incidentally doesn't work at all).

A bit of whitespace cleanup was done to stay sane, but there is lots of space/tab mixups left in there.

### CODE REVIEW

File: gpMgmt/bin/gpinitsystem [hunk 1]
```diff
@@ -136,7 +136,6 @@ USAGE () {
 		$ECHO "      -s, standby_hostname [optional]"
 		$ECHO "      -P, standby_port [optional]"
 		$ECHO "      -F, standby_filespaces [optiona]"
-		$ECHO "      -i, do not initialize standby sync process [default:- start process]"
```
Comment (larham) [inline gpMgmt/bin/gpinitsystem]: Nice find! (The actual option seems to have been removed without removing its documentation via help)

File: gpMgmt/bin/lib/gpcreateseg.sh [hunk 2]
```diff
@@ -253,16 +245,15 @@ START_QE() {
...
-		v ) VERSION_INFO ;;
-		'?' ) USAGE ;;
 		q ) unset VERBOSE ;;
-	        p ) PG_CONF_ADD_FILE=$OPTARG
+		p ) PG_CONF_ADD_FILE=$OPTARG
 		    shift
 		    shift ;;
-		* ) USAGE 
+		\? ) echo "Invalid option: -$OPTARG"
+			 exit $EXIT_STATUS ;;
```
Comment (Chibin) [inline gpMgmt/bin/lib/gpcreateseg.sh]: Should exit status be 1?

File: gpMgmt/bin/lib/gp_bash_functions.sh [hunk 3]
```diff
@@ -284,24 +225,6 @@ IN_ARRAY () {
     return 0
 }
 
-CHK_DB_QD_SET () {
-		if [ ! $MASTER_DATA_DIRECTORY ]; then
-				ERROR_EXIT "[FATAL]:-MASTER_DATA_DIRECTORY parameter not set, update user start-up file [i.e. .bashrc]." 2
-		fi
-}
-
-VERSION_INFO () {
```
Comment (larham) [inline gpMgmt/bin/lib/gp_bash_functions.sh]: Furthermore, would all utilities that include the gp_bash_functions have their 'version' option ('-v') removed?  @skahler-pivotal

Comment (skahler-pivotal) [inline gpMgmt/bin/lib/gp_bash_functions.sh]: I think version information would be good to have. Though it looks like it is broken currently
```

[gpadmin@ip-172-31-30-43 bin]$ gpinitsystem -v
gpinitsystem Version   =
gpinitsystem Status    = Released
gpinitsystem Update    =
gpinitsystem Check sum = 4221921586 82107
```

**Comment by Eulerizeit** [comment]
Is this yet another way of getting the version info? Can we replace this
with one of the numerous, it seems at least, ways that currently work?

Rob

On May 2, 2017 8:19 PM, "Scott Kahler" <notifications@github.com> wrote:

*@skahler-pivotal* commented on this pull request.
------------------------------

In gpMgmt/bin/lib/gp_bash_functions.sh
<https://github.com/greenplum-db/gpdb/pull/2351#discussion_r114454765>:

> @@ -284,24 +225,6 @@ IN_ARRAY () {
     return 0
 }

-CHK_DB_QD_SET () {
-		if [ ! $MASTER_DATA_DIRECTORY ]; then
-				ERROR_EXIT "[FATAL]:-MASTER_DATA_DIRECTORY parameter not set,
update user start-up file [i.e. .bashrc]." 2
-		fi
-}
-
-VERSION_INFO () {

I think version information would be good to have. Though it looks like it
is broken currently


[gpadmin@ip-172-31-30-43 bin]$ gpinitsystem -v
gpinitsystem Version   =
gpinitsystem Status    = Released
gpinitsystem Update    =
gpinitsystem Check sum = 4221921586 82107

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/greenplum-db/gpdb/pull/2351#discussion_r114454765>, or mute
the thread

**Comment by danielgustafsson** [comment]
Thanks everyone! Addressed all review comments and added a new way of printing the version by pulling it from the canonical source at compile time.

**Comment by Chibin** [comment]
@mkiyama Tagging as there are some doc changes.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks!

**Pull request closed without merging**